lighthouse-service/event_handler/common.go

```diff
@@ -211,16 +210,31 @@ type SLIProviderConfig interface {
 }
 
 // K8sSLIProviderConfig godoc
-type K8sSLIProviderConfig struct{}
+type K8sSLIProviderConfig struct {
+	KubeAPI kubernetes.Interface
+}
 
-// GetDefaultSLIProvider godoc
-func (K8sSLIProviderConfig) GetDefaultSLIProvider() (string, error) {
-	kubeAPI, err := GetConfig().GetKubeAPI()
-	if err != nil {
-		return "", err
+type K8sSLIProviderOption func(c *K8sSLIProviderConfig)
+
+func WithKubeAPIClient(k8sClient kubernetes.Interface) K8sSLIProviderOption {
+	return func(c *K8sSLIProviderConfig) {
+		c.KubeAPI = k8sClient
 	}
+}
 
-	configMap, err := kubeAPI.CoreV1().ConfigMaps(namespace).Get(context.TODO(), "lighthouse-config", v1.GetOptions{})
+func NewSLIProviderConfig(opts ...K8sSLIProviderOption) K8sSLIProviderConfig {
```

Comment:
we can pass the k8s interface directly as an argument since this component does not function without any k8s client injected into it